adi/ad937x.py

```diff
@@ -170,44 +170,60 @@ def tx_rf_bandwidth(self):
     @property
     def rx_enable_dec8(self):
         """rx_enable_dec8: Enable x8 decimation filter in RX path"""
-        avail = self._get_iio_attr_str(
-            "voltage0_i", "sampling_frequency_available", False, self._rxadc
-        )
-        avail = avail.strip().split(" ")
         val = self._get_iio_attr_str(
             "voltage0_i", "sampling_frequency", False, self._rxadc
         )
-        return val == avail[1]
+        avail = None
+        try:
+            avail = self._get_iio_attr_str(
+                "voltage0_i", "sampling_frequency_available", False, self._rxadc
+            )
+            avail = avail.strip().split(" ")
+            return val == avail[1]
+        except KeyError:
+            return val == avail
```

Comment:
This applies to the other blocks that share this logic.

1. If sampling_frequency_available does not exist this will generate a strange error. Something like sampling_frequency_available is not a valid key, which won't make any sense. So there needs to be an except catch for that case
2. The case when it does but only 1 value exists doesn't really make sense or the logic flow is convoluted. If only 1 value exists then it should always return false.

Reply:
I updated the implementation. It checks first if sampling_frequency_available exists then returns False if not. It it exists and only 1 value is available, it sets to False. And if it has 2 values available it returns True.